tcell_driver.go

```diff
@@ -96,6 +97,14 @@ func (g *Gui) tcellInitSimulation() error {
 // content (rune) and attributes using provided OutputMode
 func tcellSetCell(x, y int, ch rune, fg, bg Attribute, outputMode OutputMode) {
 	st := getTcellStyle(oldStyle{fg: fg, bg: bg, outputMode: outputMode})
+	if runtime.GOOS == "windows" {
+		// Dirty workaround for Windows.
```

Comment:
Do we know why we need to do this @Ryooooooga ? Is it an issue with the tcell library?

Reply:
@jesseduffield 
If I understand correctly, the cause of this problem is complex.

The root cause of the afterimage is (probably) the Windows font renderer.

If you run the following code in Powershell on Windows, you will get afterimages.

```sh
PS> echo "あいう`n あいう`rabc"
```

![afterimage](https://user-images.githubusercontent.com/10097437/215331204-1e8d9874-4716-43f4-b960-77dec635b5c0.png)

`い` in the second line is an afterimage. Thus, overwriting only the left half of the wide text leaves an afterimage on the right half.
This also occurred on different terminal emulators such as Alacritty and Windows Terminal.

If you run similar code on a mac, there will be no afterimage.

```sh
$ printf "あいう\nあいう\rabc\n"
```

<img width="274" alt="スクリーンショット 2023-01-29 22 22 12" src="https://user-images.githubusercontent.com/10097437/215331429-3bc86a55-79d3-45f4-8e99-91506c77a11e.png">

If we insert an extra space after ``c``, the afterimage disappears.

```sh
PS> echo "あいう`nあいう`rabc "
```

![c](https://user-images.githubusercontent.com/10097437/215331490-9f5e12c8-8404-467a-9739-633f1e1cf30e.png)

Therefore, when tcell or gocui overwrites the left side of a wide character with a half-width character, the right half must be cleared with a space (or other characters).

BTW, gocui clears the entire view with spaces when it re-renders the `View`.

https://github.com/jesseduffield/gocui/blob/a0bbbd86ba608ff812a752c3c8252386838fb48e/view.go#L1078

So why are there afterimages left behind?

The reason is due to the tcell mechanism. tcell skips drawing non-dirty cells.

https://github.com/gdamore/tcell/blob/5e9dd83b528187000fbf64e0dbcf582cc385ded0/console_win.go#L999

So the flow that leaves afterimages is as follows.

```
1. gocui.View#clearRunes()
cells |SP|SP|SP|SP|SP|SP|
dirty |? |? |? |? |? |? |

2. write contents "あいう"
cells |あ|SP|い|SP|う|SP|
dirty |T |F |T |F |T |F |

3. gocui.View#clearRunes()
cells |SP|SP|SP|SP|SP|SP|
dirty |T |F |T |F |T |F |
          ↑     ↑     ↑ afterimage
```

This patch works as follows.

```
1. gocui.View#clearRunes()
cells |SP|SP|SP|SP|SP|SP|
dirty |? |? |? |? |? |? |

2. write contents "あいう"
cells |あ|-1|い|-1|う|-1|
dirty |T |T |T |T |T |T |

3. gocui.View#clearRunes()
cells |SP|SP|SP|SP|SP|SP|
dirty |T |T |T |T |T |T |
       no afterimage
```

Another solution is to call `tcell.Screen#Clear` on redraw, but this will cause flickering, which will result in a bad UX.

https://github.com/gdamore/tcell/blob/5e9dd83b528187000fbf64e0dbcf582cc385ded0/console_win.go#L1159